### PR TITLE
Add Survey Statistics Feature with Comprehensive Data Visualization

### PR DESCRIPTION
**Pull request by Fovty**
## Motivation

This PR introduces a new survey statistics feature that provides users with comprehensive data visualization capabilities for analyzing survey responses. The feature enables users to view detailed statistics for different question types, including averages, medians, distributions, and segment-level analysis for target surveys.

By implementing this feature, we empower users to gain valuable insights from their survey data through intuitive visualizations and statistical summaries, enhancing the overall utility of the feedback system.

## Changes

- Created new `SurveyStatisticsController` to handle survey statistics retrieval
- Implemented `calculateStatisticsForSurvey` method in `SurveyService` to process different question types
- Added specialized handling for target survey templates with segment-level statistics
- Created a new statistics view with visualizations for different question types:
  - Range questions: Average, median, and distribution bars
  - Checkbox questions: Option selection counts
  - Text/textarea questions: Response counts
  - Target questions: Segment ratings with vertical bar visualizations
- Added a new "Statistics" link in the survey overview page
- Implemented authorization checks to ensure users can only view statistics for their own surveys
- Enhanced the `Feedback` model with a `results()` relationship to simplify data access
- Added proper error handling and fallbacks for edge cases

## Tests done

- Manually tested statistics generation with various survey datasets
- Verified correct calculation of averages, medians, and distributions
- Tested visualization of different question types (range, checkbox, text, target)
- Confirmed proper handling of edge cases (no responses, invalid data)
- Tested authorization to ensure users can only access their own survey statistics
- Verified responsive design across different screen sizes

## TODO

- [x] I've assigned myself to this PR

### CODE REVIEW

File: resources/views/livewire/surveys/overview.blade.php [hunk 1]
```diff
@@ -29,9 +29,14 @@
                         <span class="text-sm text-gray-500 dark:text-gray-400">
                             {{ $survey->already_answered }} / {{ $survey->limit == -1 ? '∞' : $survey->limit }} {{__('surveys.responses')}}
                         </span>
-                        <a href="{{ route('surveys.edit', ['id' => $survey->id]) }}" class="text-blue-500 hover:text-blue-600 text-sm">
-                            {{__('surveys.edit')}} →
-                        </a>
+                        <div class="flex gap-2">
+                            <a href="{{ route('surveys.statistics', ['survey' => $survey->id]) }}" class="text-green-500 hover:text-green-600 text-sm">
+                                {{__('Statistics')}} →
```
Comment (techmaved) [inline resources/views/livewire/surveys/overview.blade.php:34]: missing translation

**Review by techmaved — APPROVED** [review]
